c/meterpreter/source/extensions/stdapi/server/net/socket/tcp_server.c

```diff
@@ -319,29 +382,46 @@ DWORD request_net_tcp_server_channel_open(Remote * remote, Packet * packet)
 				// support IPv4 and IPv6 we'd need to create two separate sockets. IPv6 on XP isn't that common
 				// so instead, we'll just revert back to v4 and listen on that one address instead.
 				closesocket(ctx->fd);
-				v4Fallback = TRUE;
+				ai_family = AF_INET;
+				ctx->fd = WSASocket(ai_family, SOCK_STREAM, IPPROTO_TCP, 0, 0, 0);
 			}
 		}
+		if (ctx->fd == INVALID_SOCKET) {
+			BREAK_ON_WSAERROR("[TCP-SERVER] request_net_tcp_server_channel_open. WSASocket failed");
+		}
 
 		struct sockaddr_in6 sockAddr = { 0 };
 		DWORD sockAddrSize = 0;
 
-		if (v4Fallback)
+		if (ai_family == AF_INET)
 		{
 			struct sockaddr_in* v4Addr = (struct sockaddr_in*)&sockAddr;
-			v4Addr->sin_addr.s_addr = localHost == NULL ? htons(INADDR_ANY) : inet_addr(localHost);
+			if (localHost) {
+				inet_pton(AF_INET, localHost, &v4Addr->sin_addr);
+			}
+			else {
+				v4Addr->sin_addr.s_addr = INADDR_ANY;

```

Comment:
Shouldn't this be `htons(INADDR_ANY)` as this field requires a host string vs a number?

Reply:
Yeah the byte order needs to be switched. There are no strings here though, that's host-to-network-short. It doesn't really matter because `INADDR_ANY` is 0.